File: sdk/.net/ardvro.sdk.net/component/sqljson/controller/DataExportController.cs
namespace ardvro.sdk.net.component.sqljson.controller;

using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class DataExportController
{
    private IConnection _conection;

    public DataExportController(IConnection connection)
    {
        _conection = connection;
    }

    public void ExportCsv(String countryCode, Object data, String template, int width, int height, Action<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/DataExport/ExportCsv";
        _conection.Submit(new Object[]{ countryCode, data, template, width, height }, functionName, onResponse);
    }

    public void ExportExcel(String countryCode, Object data, String template, int width, int height, Action<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/DataExport/ExportExcel";
        _conection.Submit(new Object[]{ countryCode, data, template, width, height }, functionName, onResponse);
    }

    public void ExportHtml(String countryCode, Object data, String template, int width, int height, Action<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/DataExport/ExportHtml";
        _conection.Submit(new Object[]{ countryCode, data, template, width, height }, functionName, onResponse);
    }

    public void ExportPdf(String countryCode, Object data, String template, int width, int height, Action<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/DataExport/ExportPdf";
        _conection.Submit(new Object[]{ countryCode, data, template, width, height }, functionName, onResponse);
    }

    public void ExportTxt(String countryCode, Object data, String template, int width, int height, Action<byte[]> onResponse)
    {
        String functionName = "ardvro/component/sqljson/DataExport/ExportTxt";
        _conection.Submit(new Object[]{ countryCode, data, template, width, height }, functionName, onResponse);
    }

}

File: sdk/web/component/sqljson/controller/DataExportController.js
var DataExportController = function DataExportController(cfg)
{
    let ctrl = {};

    ctrl.ExportCsv = function (countryCode, data, template, width, height, callback)
    {
        var item = {
            CountryCode: countryCode,
            Data: data,
            Temlate: template,
            Width: width,
            HeighT: height
        };

        cfg.Connector.Submit(item, "ardvro/component/sqljson/DataExport/ExportCsv", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.ExportExcel = function (countryCode, data, template, width, height, callback)
    {
        var item = {
            CountryCode: countryCode,
            Data: data,
            Temlate: template,
            Width: width,
            HeighT: height
        };

        cfg.Connector.Submit(item, "ardvro/component/sqljson/DataExport/ExportExcel", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.ExportHtml = function (countryCode, data, template, width, height, callback)
    {
        var item = {
            CountryCode: countryCode,
            Data: data,
            Temlate: template,
            Width: width,
            HeighT: height
        };

        cfg.Connector.Submit(item, "ardvro/component/sqljson/DataExport/ExportHtml", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.ExportPdf = function (countryCode, data, template, width, height, callback)
    {
        var item = {
            CountryCode: countryCode,
            Data: data,
            Temlate: template,
            Width: width,
            HeighT: height
        };

        cfg.Connector.Submit(item, "ardvro/component/sqljson/DataExport/ExportPdf", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.ExportTxt = function (countryCode, data, template, width, height, callback)
    {
        var item = {
            CountryCode: countryCode,
            Data: data,
            Temlate: template,
            Width: width,
            HeighT: height
        };

        cfg.Connector.Submit(item, "ardvro/component/sqljson/DataExport/ExportTxt", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    return ctrl;
};